### PR TITLE
Tony update arvatar

### PR DESCRIPTION
**Pull request by nguyenductoan**
#17 , #18

### CODE REVIEW

**Comment by nguyenductoan** [comment]
@minhquan4080 please review my code.

File: app/controllers/studio/art_pieces_controller.rb [hunk 1]
```diff
@@ -9,25 +9,44 @@ def index
...
     @art_piece.user = current_user
     
     @art_piece.approved = true
 
     set_artist_id
-
-    if @art_piece.save  
-      redirect_to studio_art_pieces_path
+    if @art_piece.save
+      byebug
```
Comment (minhquan4080) [inline app/controllers/studio/art_pieces_controller.rb]: remove

File: app/controllers/studio/art_pieces_controller.rb [hunk 2]
```diff
@@ -9,25 +9,44 @@ def index
 
 
   def new
+    session[:image_list] = 'nil'
     @art_piece = ArtPiece.new
   end
 
   def show
```
Comment (minhquan4080) [inline app/controllers/studio/art_pieces_controller.rb]: def show;end

File: app/controllers/studio/art_pieces_controller.rb [hunk 3]
```diff
@@ -9,25 +9,44 @@ def index
 
 
   def new
+    session[:image_list] = 'nil'
```
Comment (minhquan4080) [inline app/controllers/studio/art_pieces_controller.rb]: session[:image_list] = nil

File: app/models/gallery.rb [hunk 4]
```diff
@@ -11,6 +11,10 @@ class Gallery < ActiveRecord::Base
   
   scope :public_galleries, -> { order(created_at: :desc).where(approved: true) }
   
+  before_create do
+    self.approved = true
+  end
+  
   def self.featured
     Gallery.where(featured: true).public_galleries.limit(1)
```
Comment (minhquan4080) [inline app/models/gallery.rb]: Gallery.where(featured: true).public_galleries.take

File: app/views/studio/art_pieces/new.html.slim [hunk 5]
```diff
@@ -1,5 +1,5 @@
 - breadcrumb :studio_uploads
 .container
   = form_for @art_piece, :url => studio_art_pieces_path, :method => :post, :html => { :class => "art-piece-upload-form", :multipart => true } do |f|
-    = render "form", f:f
+    = render "form1", f:f
```
Comment (lhhieu89) [inline app/views/studio/art_pieces/new.html.slim]: What are differences between `form` and `form1`?

File: app/views/up_load_panting_mailer/to_admin.erb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+Dear Admin,
+<%= @user.name %> has already updated new panting with title <%= @art_piece.title %>.
```
Comment (lhhieu89) [inline app/views/up_load_panting_mailer/to_admin.erb]: `panting` or `painting`?
Please change the file name, too

File: app/mailers/up_load_panting_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class UpLoadPantingMailer < ActionMailer::Base
+  default from: "admin@artfolo.com"
+  def to_admin(art_piece)
+    @user = art_piece.user
+    @art_piece = art_piece
+    subject = "New panting uploaded"
+    mail(to: User.admins.pluck(:email), subject: subject, template_name: 'to_admin')
+  end
+end
```
Comment (lhhieu89) [inline app/mailers/up_load_panting_mailer.rb]: Change the text and file name from `panting` to `painting` please

File: app/controllers/studio/art_pieces_controller.rb [hunk 8]
```diff
@@ -9,25 +9,43 @@ def index
...
+    new_image_list = params[:art_piece_image_tmp]
+    #if new_image_list[-1, 1] != ','
+    #  new_image_list = art_piece_params[:image] + ','
+    #  image_list = art_piece_params[:image] + ','
+    #end
+    url = new_image_list.split(',')
+    if url.length > 0
+      @art_piece[:image] = url[0]
+      new_image_list = new_image_list.gsub(url[0] + ',', '')
+      session[:image_list] = new_image_list
```
Comment (lhhieu89) [inline app/controllers/studio/art_pieces_controller.rb]: You can rewrite
```ruby
new_image_list = new_image_list.gsub(url[0] + ',', '')
session[:image_list] = new_image_list
```
to
```ruby
url.shift
session[:image_list] = url.join(',')
```

**Pull request merged**